Sources/SwiftCompilerPluginMessageHandling/PluginMacroExpansionContext.swift

```diff
@@ -186,6 +186,23 @@ class SourceManager {
     )
   }
 
+  func range(
+    of node: some SyntaxProtocol,
+    localRange: Range<AbsolutePosition>
+  ) -> SourceRange? {
```

Comment:
Could you add a doc comment to this method? 

Also: Could `range(of:from:to:)` call into this method instead of duplicating logic?